src/Http/Http.Results/test/ResultsOfTHelperTests.cs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.DotNet.RemoteExecutor;
+
+namespace Microsoft.AspNetCore.Http.HttpResults;
+
+public class ResultsOfTHelperTests
+{
+    [ConditionalTheory]
+    [RemoteExecutionSupported]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void PopulateMetadataIfTargetIsIEndpointMetadataProvider_PublicMethod_Called(bool isDynamicCodeSupported)
+    {
+        var options = new RemoteInvokeOptions();
+        options.RuntimeConfigurationOptions.Add("System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported", isDynamicCodeSupported.ToString());
+
+        using var remoteHandle = RemoteExecutor.Invoke(static () =>
+        {
+            var methodInfo = typeof(ResultsOfTHelperTests).GetMethod(nameof(PopulateMetadataIfTargetIsIEndpointMetadataProvider_PublicMethod_Called));
+            var endpointBuilder = new TestEndpointBuilder();
+
+            ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider<PublicMethodEndpointMetadataProvider>(
+                methodInfo,
+                endpointBuilder);
+
+            Assert.Single(endpointBuilder.Metadata);
```

Comment:
I wonder if we could reduce the boilerplate/copied code by just combining these 3 tests into one, and calling ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider 3 times in one test method.

Reply:
I don't want to merge them. However, I've reduced the amount of duplication with a helper method.